lib/src/parallel_prefix_operations.dart

```diff
@@ -0,0 +1,250 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// parallel-prefix_operations.dart
+// Implementation of operators using various parallel-prefix trees.
+//
+// 2023 Sep 29
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+//   Borrowed from https://github.com/stevenmburns/rohd_sklansky.git
+
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// This computes the power of 2 less than x
+int largestPow2LessThan(int x) => pow(2, log2Ceil(x) - 1).toInt();
+
+/// ParallePrefix is the core parallel prefix tree structure
+class ParallelPrefix extends Module {
+  final List<Logic> _oseq = [];
+
+  /// Output sequence value
+  List<Logic> get val => _oseq;
+
+  /// ParallePrefix recursion
+  ParallelPrefix(List<Logic> inps, String name) : super(name: name) {
```

Comment:
In general, is there a reason why we use `List<Logic>` where each is 1-bit instead of just a single `Logic` bus with varying width?  It seems like it would be easier to use that way as well

Reply:
Each of these items in the List are a multi-bit Logic (2).  See line 202, where the basic operations or AND and OR of the input bits, swizzled into a two-bit representation to feed into the tree.  So each input of the tree is a two-bit Logic and this carries through the core computation.  It might be more clear if this 'Logic' was declared as a type that carried just two bits.

Reply:
Ah I see, ok that makes sense to me.  Maybe a little more explanation on the doc comment for `ParallelPrefix` could help elaborate why it's a list and what sorts of things are expected?

Reply:
Would it be crazy to create a local two-bit Logic type?

Reply:
Are all use cases of prefix trees always 2-bit?

If we want `ParallelLogic` to be an API, we would need to consider usage of the API.  Do we want people to have to construct a special type of `Logic` in order to use the API?

Reply:
I know the Logic type is a single length, typically 2, but there are longer variants.  These are things like carry-save, generate-propagate or generate-kill.  I found a 4-bit version, for example.  So ParallelPrefix is really just a node in a tree of these, which transform N k-bit quantities in a row to another N k-bit quantities in the next row to form a tree.